src/mixpanel_utils/__init__.py

```diff
@@ -106,17 +106,23 @@ def __init__(
         self.pool_size = pool_size
         self.read_pool_size = read_pool_size
         self.max_retries = max_retries
-        self.eu = eu
+        self.residency = residency
+        residency_values = ["us","eu","in"]
+        assert self.residency in residency_values, "residency value must be 'us', 'eu, or 'in'!"
         self.raw_api = (
-            "https://data.mixpanel.com/api"
-            if eu is False
-            else "https://data-eu.mixpanel.com/api"
+            "https://data.mixpanel.com/api" if residency == "us"
+            else "https://data-eu.mixpanel.com/api" if residency == "eu"
+            else "https://data-in.mixpanel.com/api"
```

Comment:
let's follow what Ted suggested and make this future proof. The validation logic ensures we will have valid residency value so we can do something like 
if us:  "https://data.mixpanel.com/api"
else: f"https://data-{residency}.mixpanel.com/api"